rotkehlchen/exchanges/binance.py

```diff
@@ -1310,6 +1320,99 @@ def _query_online_fiat_payments(
 
         return events
 
+    def _query_online_convert_trades(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> list[SwapEvent]:
+        """Query Binance Convert trades using the convert/tradeFlow endpoint.
+        Docs: https://developers.binance.com/docs/convert/trade/Get-Convert-Trade-History
+        """
+        convert_trades = self._api_query_list_within_time_delta(
+            start_ts=start_ts,
+            end_ts=end_ts,
+            time_delta=CONVERT_API_TIME_DELTA,
+            api_type='sapi',
+            method='convert/tradeFlow',
+            additional_options={'limit': CONVERT_API_MAX_LIMIT},
+        )
+        log.debug(f'{self.name} convert trades history result', results_num=len(convert_trades))
+
+        events = []
+        for raw_convert in convert_trades:
+            events.extend(self._deserialize_convert_trade(raw_data=raw_convert))
+
+        return events
+
+    @staticmethod
+    def handle_deserialization_errors(
```

Comment:
Not for this PR. But this kind of decorator can be used in many many more places/exchanges if generalized well enough. Can you make a note for this?